pkg/build/strategies/sti/sti.go

```diff
@@ -296,6 +296,13 @@ func (b *STI) Save(request *api.Request) (err error) {
 func (b *STI) Execute(command string, request *api.Request) error {
 	glog.V(2).Infof("Using image name %s", request.BaseImage)
 
+	env, err := scripts.GetEnvironment(request)
+	if err != nil {
+		glog.V(1).Infof("No .sti/environment provided (%v)", err)
+	}
+
+	buildEnv := append(scripts.ConvertEnvironment(env), b.generateConfigEnv()...)
```

Comment:
good catch, we should do the same as for on build. I will add that tomorrow.
On Feb 24, 2015 6:48 PM, "Ben Parees" <notifications@github.com> wrote:

> In pkg/build/strategies/sti/sti.go
> <https://github.com/openshift/source-to-image/pull/145#discussion_r25273877>
> :
>
> > @@ -296,6 +296,13 @@ func (b *STI) Save(request *api.Request) (err error) {
> >  func (b *STI) Execute(command string, request *api.Request) error {
> >  	glog.V(2).Infof("Using image name %s", request.BaseImage)
> >
> > +	env, err := scripts.GetEnvironment(request)
> > +	if err != nil {
> > +		glog.V(1).Infof("No .sti/environment provided (%v)", err)
> > +	}
> > +
> > +	buildEnv := append(scripts.ConvertEnvironment(env), b.generateConfigEnv()...)
>
> what happens for layered builds (the case where we don't have tar)?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/source-to-image/pull/145/files#r25273877>.
>